test/unit/typescript.spec.ts

```diff
@@ -0,0 +1,40 @@
+/*!
+ * @license
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import * as chai from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+
+import { messaging } from '../../src/messaging/index';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+const expect = chai.expect;
+
+describe('Typescript', () => {
+  // This test was added as part of https://github.com/firebase/firebase-admin-node/issues/1146
+  it('should export all types of Messages so it can be used by lib users', () => {
```

Comment:
Let's try to incorporate this test logic to the existing `messaging.spec` file. No need to carve out a new file for this. Easiest would be to list it under the existing test suite for the `send()` method:

```typescript
      it(`should be fulfilled with a message ID given a TokenMessage`, () => {
        const message: TokenMessage = {token: 'test'}
        mockedRequests.push(mockSendRequest());
        return messaging.send(message)
          .should.eventually.equal('projects/projec_id/messages/message_id');
      });

     // Add similar test cases for other message types.
```

Also note that when you add a new test file, you have to also list it in `index.spec.ts`. Otherwise mocha won't run it. But, let's just add this to the existing unit test file.

Reply:
Or alternatively you can have one test case in the suite for `sendAll()`:

```typescript
    it('should be fulfilled when called with different message types', () => {
      const messageIds = [
        'projects/projec_id/messages/1',
        'projects/projec_id/messages/2',
        'projects/projec_id/messages/3',
      ];
      const tokenMessage: TokenMessage = {token: 'test'};
      const topicMessage: TopicMessage = {topic: 'test'};
      const conditionMessage: ConditionMessage = {condition: 'test'};
      const messages: Message[] = [tokenMessage, topicMessage, conditionMessage];
      mockedRequests.push(mockBatchRequest(messageIds));
      return messaging.sendAll(messages)
        .then((response: BatchResponse) => {
          expect(response.successCount).to.equal(3);
          expect(response.failureCount).to.equal(0);
          response.responses.forEach((resp, idx) => {
            expect(resp.success).to.be.true;
            expect(resp.messageId).to.equal(messageIds[idx]);
            expect(resp.error).to.be.undefined;
          });
        });
    });
```

Reply:
Makes sense. Thx for the guidance!

---

> Also note that when you add a new test file, you have to also list it in index.spec.ts. Otherwise mocha won't run it

FYI mocha _did_ run the new test. I also verified when I added the new file 🙂 

<img width="612" alt="Screenshot 2021-02-04 at 08 43 04" src="https://user-images.githubusercontent.com/7549295/106860678-0269a300-66c5-11eb-85c2-cb494afe8231.png">

Source: https://github.com/firebase/firebase-admin-node/runs/1820854315